src/Instrumentation/Psr18/src/Psr18Instrumentation.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Instrumentation\Psr18;
+
+use function get_cfg_var;
+use OpenTelemetry\API\Common\Instrumentation;
+use OpenTelemetry\API\Common\Instrumentation\CachedInstrumentation;
+use OpenTelemetry\API\Trace\Span;
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\StatusCode;
+use OpenTelemetry\Context\Context;
+use function OpenTelemetry\Instrumentation\hook;
+use OpenTelemetry\SemConv\TraceAttributes;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use function sprintf;
+use function strtolower;
+use Throwable;
+
+class Psr18Instrumentation
+{
+    /** @psalm-suppress ArgumentTypeCoercion */
+    public static function register(): void
+    {
+        hook(
+            ClientInterface::class,
+            'sendRequest',
+            pre: static function (ClientInterface $client, array $params, string $class, string $function, ?string $filename, ?int $lineno): ?array {
+                $request = $params[0] ?? null;
+                if (!$request instanceof RequestInterface) {
+                    Context::storage()->attach(Context::getCurrent());
+
+                    return null;
+                }
+
+                static $instrumentation;
```

Comment:
Can we move this instance to `register` function and follow the same pattern as for PSR-15?